### ISSUE TITLE
Custom column filter allows filtering only by number

### ISSUE THREAD

**Issue by easeev** [metabase/metabase#13751]
**Describe the bug**
Filter by custom column allows filtering only by number

**To Reproduce**
1. Create new question
2. Create custom column by using `regexextract()`
3. Create a filter using custom column
4. See that filter works only with numbers

**Expected behavior**
Filter works with strings

**Screenshots**

![image](https://user-images.githubusercontent.com/14873170/98762267-3a3cb380-2412-11eb-9d5d-cfea7372e88b.png)

**Information about your Metabase Installation:**

```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.183 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "ANSI_X3.4-1968",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "1.8.0_272-8u272-b10-0+deb9u1-b10",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_272",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "25.272-b10",
    "os.name": "Linux",
    "os.version": "5.4.49+",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.4"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v1.37.0.2",
      "date": "2020-10-26",
      "branch": "release-x.37.x",
      "hash": "7996551"
    },
    "settings": {
      "report-timezone": "Asia/Kuala_Lumpur"
    }
  }
}
```

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by flamber** [comment]
Related to #13217 and #13122

**Comment by nemanjaglumac** [comment]
I was trying to write a repro for this issue but cannot get past `No method in multimethod '->honeysql' for dispatch value: [:h2 :regex-match-first]` when trying to add a custom column.
![image](https://user-images.githubusercontent.com/31325167/99856232-cffdde80-2b88-11eb-9ad0-4c35668a8d99.png)

I saw that in #13634. Is it safe to assume that this issue is blocked by 13634 or am I doing something wrong here? cc @flamber

**Comment by flamber** [comment]
@nemanjaglumac H2 doesn't support regex. And you shouldn't be able to input that, but that's 12268/12222.

**Comment by nemanjaglumac** [comment]
Oh, good to know. We now have mysql, postres and mongo so I can try on any of those? Which one supports regex?

**Comment by flamber** [comment]
@nemanjaglumac Postgres: https://www.metabase.com/docs/latest/users-guide/expressions.html#database-limitations - Mongo doesn't support Custom Expressions at all.

**Comment by nemanjaglumac** [comment]
> @nemanjaglumac Postgres: https://www.metabase.com/docs/latest/users-guide/expressions.html#database-limitations - 

Thank you. I could've looked at the log before I asked that question. My bad. Sorry.
```
"application-database": "postgres",
```

**Comment by jeff303** [comment]
This issue also applies to `replace`, which suggests there isn't something specifically wrong with `regexextract`.

**Comment by flamber** [comment]
Fixed by the work in #15952
